Dockerfile

```diff
@@ -1,23 +1,16 @@
-# Use Python 3.11 base image
-FROM python:3.11-slim
+# Stage 1: Build dependencies and run tests
+FROM python:3.11-slim AS builder
 
 # Set environment variables
-
-# This environment variable instructs Python not to write bytecode files (.pyc) to disk.
-# Bytecode files are typically used to speed up subsequent imports of the same module,
-# but in containerized environments or when filesystems are considered ephemeral,
-# writing bytecode files can be unnecessary and may cause issues.
 ENV PYTHONDONTWRITEBYTECODE 1
-
-# This environment variable ensures that Python's standard output (stdout) and standard error (stderr)
-# streams are unbuffered, meaning that the output is immediately flushed to the output stream.
-# In containerized environments, buffering can cause delays in output being displayed or logged,
-# so setting this variable to a non-empty value (typically '1') helps ensure prompt and correct output behavior.
 ENV PYTHONUNBUFFERED 1
 
 # Set working directory in the container
 WORKDIR /app
 
+# Create a non-privileged user
+RUN adduser --disabled-password --gecos '' appuser
```

Comment:
GECOS in Linux refers to a field in the user database that holds information about a user, such as their full name, contact details, or other descriptive information. This field is typically found in the /etc/passwd file and is used to provide additional context about users on the system.

The empty string '' means that no additional information like full name or contact details is provided for the user.